### PR TITLE
Fix to PeriodicPadding to match pytorch circular padding

### PR DESCRIPTION
**Pull request by jvwilliams23**
Hi,

I found that the output of `lm.PeriodicPadding2D` did not match the output of `torch.nn.functional.pad(*, mode="circular")`. See below for figure comparing the original implementation - clearly the middle image is doing the equivalent of `mode = "replicate"` padding (not periodic/circular). I modified the existing CI test to compare against the torch function (it previously compared against padding done using np.concatenate).

lbann_periodicpadding_error

I also made some modifications to the docstring.

Cheers,
Josh

### CODE REVIEW

**Review by fiedorowicz1 — COMMENTED** [review]
Great catch, thanks for the PR! There are a couple of other things I caught that I will add to this PR.

**Review by fiedorowicz1 — APPROVED** [review]

**Review by bvanessen — APPROVED** [review]
